azure-quantum/azure/quantum/target/ionq.py

```diff
@@ -103,13 +106,28 @@ def submit(
             )
         if input_params is None:
             input_params = {}
+
+        num_shots = kwargs.pop("num_shots", None)
+
         if num_shots is not None:
+            warn(
+                "The 'num_shots' will be deprecated. Please use 'shots' instead.",
+                category=DeprecationWarning,
+            )
+            shots = num_shots
```

Comment:
nit: I think, I might've confused you earlier when we were discussing what happens when both `shots` and `num_shots` are specified...
Now I think that since we're deprecating the `num_shots`, the `shots` should always take precedence if `num_shots` and `shots` are specified. Could you please make this change?

Reply:
Yeah, sorry. Now that case has to be handled.